### ISSUE TITLE
[java] Possible False Positive (Unnecessary wrapper)

### ISSUE THREAD

**Issue by davidgoate** [pmd/pmd#1393]
**Affects PMD Version:**  3.9.0

**Rule:** UnnecessaryWrapperObjectCreation

**Description:** warns about `PMD.UnnecessaryWrapperObjectCreation`

**Code Sample demonstrating the issue:**

```java
    private Integer getPrice(final Stream<Integer> examples) {
        final OptionalDouble average = examples.average();
        if (average.isPresent()) {
            return Double.valueOf(average.getAsDouble()).intValue();
        } else {
            return null;
        }
    }
```

I'm not sure of another way to return `null` or the int value of the `OptionalDouble`.

**Running PMD through:** Maven

**Comment by jsotuyod** [comment]
@davidgoate the code is complaining to this line un particular:

```java
Double.valueOf(average.getAsDouble()).intValue();
```

you are taking a `double`, to create a `Double` just to get an `int`. The intermediate `Double` is pointless. As per [`Double.intValue()`'s documentation](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#intValue()), it simply casts the value to `int`, and is therefore equivalent to using:

```java
return (int) average.getAsDouble();
```
